src/multi_processing/pool_tasks.py

```diff
@@ -1,120 +1,157 @@
+    # possible using the Pool.map method
+    nested_list_of_created_records = [
+        async_result_object.get() for async_result_object in [
+            create_pool.apply_async(
+                create_records_from_create_job, args=(
+                    create_job, object_factory
+                )
+            ) for create_job in dequeued_create_jobs
+        ]
+    ]
+
+    # use a list comprehension to flatten the list of lists created above
+    created_records_from_multiple_jobs = [
+        created_record
+        for list_of_created_records in nested_list_of_created_records
+        for created_record in list_of_created_records
+    ]
+
+    create_pool.close()
+    create_pool.join()
+
+    return created_records_from_multiple_jobs
 
 
 def make_global(local_lock):
+    """ helper function used in run_create_jobs that assigns the local_lock
+    parameter to a global lock variable. This is required since a
+    multiprocessing Lock object cannot otherwise be passed to a Pool method
+    since it is not pickleable (required due to implementation of Pool in the
+    multiprocessing module).
+
+    For more information see this SO thread (with line break for PEP8):
+    https://stackoverflow.com/
+    questions/25557686/python-sharing-a-lock-between-processes
+    """
     global lock
     lock = local_lock
 
 
-def generate_data(job):
-    """ Generates a single set of records based on an individual job.
-
-    Instruction contains the number of records to generate, or the number of
-    records to retrieve from the database (in the case of dependent objects),
-    as well as the starting ID for this set in the case of sequential IDs.
-
-    Where objects are generated in nondeterministic amounts, sequential IDs
-    are not possible.
+def create_records_from_create_job(create_job, object_factory):
+    """ Returns a list of records created as specified by a single create job.
 
     Parameters
     ----------
-    job : list
-        List of 2 elements, firstly, the production instructions for objects:
-        quantity to produce and the id to start generation from. The second
-        element is the instantiated factory.
+    create_job : dict
+        dictionary specifying a quantity of records to create and the ID to
+        start from (for domain objects with sequential unique IDs)
+    object_factory : Creatable
+        Instantiated and pre-configured object factory used to create
+        records from create jobs
 
     Returns
     -------
     list
-        List containing all the records produced by this factory for this
-        instruction set.
+        List containing all the records created for this job
     """
 
-    instructions = job[0]
-    object_factory = job[1]
-
-    quantity = instructions['quantity']
-    start_id = instructions['start_id']
+    quantity, start_id = create_job['quantity'], create_job['start_id']
 
+    # this is run within the create parent process, which declared a global
```

Comment:
I think this paragraph can be trimmed down.  I think that we get all the relevant information across by saying "The InstrumentFactory is the only factory which has a critical section and therefore requires a lock"